src/Translator/Extractor/AdminExtractor.php

```diff
@@ -76,6 +76,11 @@ public function __construct(Pool $adminPool, BreadcrumbsBuilderInterface $breadc
         $this->breadcrumbsBuilder = $breadcrumbsBuilder;
     }
 
+    /**
+     * Extracts translation messages from files, a file or a directory to the catalogue.
+     *
+     * @param string|array $resource Files, a file or a directory
```

Comment:
Bonus points if you can add more specific information to the `array` using `@phpstan-param`

Reply:
`@param string|array<string>`

Reply:
How can I use phpstan to get that info? I've been seeing @phpstan-param inside the files of the project but wasn't sure what they mean (didn't used phpstan until last few days :) )

Reply:
phpstan won't give you the info. It used to validate the method calls and parameters.

IDE doesn't understand complex typehint like
```
array<int, string>
class-string
...
```
So we generally promote
- A typehint that IDE can understand for `@param`
- The full description for `@phpstan-param`.

Example
```
@param string
@phpstan-param class-string
```

```
@param string[]
@phpstan-param array<int, string>
```

In you case, `string|string[]` is the best choice, since IDE can understand it (but not `array<string>`), it avoid a `phpstan-param`.